### PR TITLE
Flatten historic

### PR DESCRIPTION
**Pull request by wilberquito**
This PR aims to find a better solution at the problem of detecting outliers. By now we had explore mean + std, median + Mae and here we explore flatten time series + median + std and flatten time series + Mae.

### CODE REVIEW

**Review by quimdt — APPROVED** [review]
Nice job :tada: 

I just make observations about the structure of the code and how to improve the readability . The algorithm is nice. May is good idea when comparing two versions of the algortihms having in separated classes.

File: src/alerts/time_series.py [hunk 1]
```diff
@@ -3,12 +3,214 @@
...
+            std_value = np.std(values)
+
+            lower_bound = mean_value - self.quantile * std_value
+            upper_bound = mean_value + self.quantile * std_value
+        else:
+            median = self._median(values)
+            mad = self._mad(values)
+
+            # Calculate bounds
+            lower_bound = median - (self.quantile * 1.4826 * mad)
```
[the remark this answers is not in the document]
Comment (wilberquito) [reply src/alerts/time_series.py:96]: @quimdt this was made by Alex HAHAH it as a is a scaling facto used when working with the mean absolute deviation (MAD).  When the dataset follows a normal distribution, the expected value of the MAD is related to the standard deviation (σ) of the distribution.

File: src/alerts/time_series.py [hunk 2]
```diff
@@ -3,12 +3,214 @@
...
 
-from typing import Any, Dict, List
+from typing import Any, Dict, List, Tuple, Optional
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from alerts.anomaly import AnomalyDetector
+
+
+class AnomalyTS():
```
[the remark this answers is not in the document]
Comment (wilberquito) [reply src/alerts/time_series.py:13]: @quimdt True

File: src/alerts/anomaly.py [hunk 3]
```diff
@@ -3,31 +3,34 @@
...
         window = ts.WINDOW
 
         for i, (date, value) in enumerate(
-            zip(ts.dates[window:], ts.values[window:]), window
+            zip(dates[window:], values[window:]), window
         ):
             # Detect and store the alert
-            if value > pred_interval["upper_bound"][i]:
-                alerts.append({"date": date, "volume": value})
+            if value > upper_bound_limits[i]:
```
Comment (oscar-delgado) [inline src/alerts/anomaly.py:33]: I don't fully understand the benefits of doing 4 loops instead of one

File: src/alerts/plot.py
Comment (oscar-delgado) [inline src/alerts/plot.py:1]: Maybe this shouldn't be in the library source since this isn't needed for the execution

File: src/alerts/time_series.py
Comment (oscar-delgado) [inline src/alerts/time_series.py:1]: The class TimeSeries is used among other projects, so its name and main functions should remain the same for not breaking compatibility with other parts of the code. Can you refactor it to do so? If you need more info about what is been used and what not, we can talk about it!

**Review by oscar-delgado — CHANGES_REQUESTED** [review]
Nice! Once the class names are reviewed, we can test it out in the code itself!

File: src/alerts/preprocess.py [hunk 4]
```diff
@@ -2,8 +2,32 @@
...
+        data (pd.DataFrame): A DataFrame containing
+            features and a "date" column.
+
+    Returns:
+        pd.DataFrame: A new DataFrame with the date set
+            as the index and features resampled
+            on a daily frequency, indicating the
+            daily count of occurrences for each feature.
+    """
+    assert "date" in data.columns
```
Comment (oscar-delgado) [inline src/alerts/preprocess.py:25]: Assert statements are reserved for testing purposes as we can not control it. Use Exception or a custom exception

File: src/alerts/dataclasses.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+from dataclasses import dataclass
+from typing import List
+
+
+@dataclass
+class LimitInterval:
+    def __init__(self, lb: float, ub: float):
+        self.lower_bound = lb
+        self.upper_bound = ub
```
Comment (quimdt) [inline src/alerts/dataclasses.py:9]: Define it better in that way, init method is used to add extra logic
```suggestion
class LimitInterval:
    lower_bound: float
    upper_bound float
```

**Pull request merged**